packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -446,6 +459,17 @@ export class GarbageCollector implements IGarbageCollector {
             && !gcOptions.disableGC
         );
 
+        const currentVersion = pkgVersion;
```

Comment:
Is back compat code required here? I don't see any issues if a client running older version reads the summary written by new version client and vice-versa.

Reply:
This is more in case I write a bug in this version, and this becomes an older client. Then we would be delayed on rollout until everyone moves off this version or consumes the next version. As this change effects summaries, I want the ability to rollback very quickly. As soon as we get a fix in for the new version, I would like to roll forward. It's to enable summary handles for future new versions, and not worry about old versions summarizing incorrectly.

Reply:
I see. It's not clear what this code is trying to do. Maybe it will help if you can move this specific logic into a separate function and add comments what is happening here with an example of how its supposed to work.